vlfs/src/driver/flash.rs

```diff
@@ -1,38 +1,84 @@
     async fn erase_sector_4kib(&mut self, address: u32) -> Result<(), Self::Error>;
     async fn erase_block_32kib(&mut self, address: u32) -> Result<(), Self::Error>;
     async fn erase_block_64kib(&mut self, address: u32) -> Result<(), Self::Error>;
 
-    // maximum read length is 4 kb
-    // size of the buffer must be at least 5 bytes larger than read_length
+    /// Reads 4KiB of data from the memory device starting at the specified address.
+    /// maximum read length is 4 kb
+    /// size of the buffer must be at least 5 bytes larger than read_length
+    /// 
+    /// #Arguments
+    /// 
+    /// * `address` - u32 integer that specifies the starting address of the 4KiB block that will be read
+    /// * `read_length` - usize integer that specifies the # of bytes to be read from the memory device.
+    /// * `read_buffer` - buffer where read data will be stored
+    /// 
+    /// #Examples
+    /// 
+    /// ```
+    /// let mut read_buffer = [0u8; 4096];
+    /// let read_data = flash.read_4kib(0x0000_0000, 4096, &mut read_buffer).await;
+    ///
+    /// ```
+    /// 
     async fn read_4kib<'b>(
         &mut self,
         address: u32,
         read_length: usize,
         read_buffer: &'b mut [u8],
-    ) -> Result<&'b [u8], Self::Error>;
-
-    // Write a full page of 256 bytes, the last byte of the address is ignored
-    // The write buffer must be less than or equals 261 bytes long, where the last 256 bytes are the data to write
+    ) -> Result<&'b [u8], Self::Error>;//
+
+
+    /// Writes 256 bytes of data to the memory device starting at the specified address.
+    /// size of the buffer must be at least 261 bytes long
+    /// 
+    /// #Arguments
+    /// 
+    /// * `address` - u32 integer that specifies the starting address of the 256 byte block that will be written
+    /// * `write_buffer` - buffer where data to be written is stored
+    /// 
+    /// #Examples
+    /// 
+    /// ```
+    /// let mut write_buffer = [0u8; 261];
+    /// flash.write_256b(0x0000_0000, &mut write_buffer).await;
+    /// 
+    /// ```
     async fn write_256b<'b>(
         &mut self,
         address: u32,
         write_buffer: &'b mut [u8],
     ) -> Result<(), Self::Error>;
 
-    // read arbitary length, length of read_buffer must be larger or equal to read_length + 5
+
+    /// Reads arbitary length of data from the memory device starting at the specified address.
+    /// maximum read length is 4 kb
+    /// size of the buffer must be at least read_length + 5 bytes long
+    /// refer to the read_4kib function for more details on the parameters and outputs as they are the same
+    /// 
+    /// This loop reads the data from the flash by calling read_4kib multiple times, with each call
+    /// reading up to 4096 bytes at a time, until it has read the entire requested length.
+    /// 
+    /// Variables:
+    ///    bytes_read: usize integer that keeps track of the number of bytes that have been read so far
```

Comment:
These are internal variables of the function, they don't need to be described here. The comments before a function should tell the user how to use the function and what will the function do, not the implementation detail of the function (unless it affects how the user would use the function). Put these comments in side the function, before the variable is defined.